addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,794 @@
+                    }
+                    break;
+
+                case CHANNEL_TIVO_COMMAND:
+                    logger.debug("handleCommand '{}' - CHANNEL_TIVO COMMAND found!", getThing().getUID());
+
+                    // Attempt to execute the command on the tivo
+                    myTivo = myTivoService.cmdTivoSend(tmpAct);
+
+                    // Check to see if the command was successful
+                    if (myTivo != null && myTivo.isCmdOk()) {
+                        logger.debug("handleCommand '{}' - returned Tivo Data Object: '{}'", getThing().getUID(),
+                                myTivo.toString());
+                        updateTivoStatus(myTivo);
+                    } else {
+                        logger.warn("handleCommand '{}' - command failed '{}'", getThing().getUID(), tmpAct);
+                    }
+                    break;
+
+                case CHANNEL_TIVO_TELEPORT:
+                    tivoCommand = "TELEPORT " + tmpAct;
+
+                    logger.debug("handleCommand '{}' TELEPORT command to tivo: '{}'", getThing().getUID(), tivoCommand);
+
+                    if (myTivoService.getServiceStatus().getConnOK() == 2 & tmpAct.equals("TIVO")) {
+                        tivoCommand = "IRCODE " + tmpAct;
+                        logger.debug("TiVo '{}' TELEPORT re-mapped to IRCODE as we are in standby: '{}'",
+                                getThing().getUID(), tivoCommand);
+                    }
+
+                    // Attempt to execute the command on the TiVo
+                    myTivo = myTivoService.cmdTivoSend(tivoCommand);
+
+                    // Check to see if the command was successful
+                    if (myTivo != null && myTivo.isCmdOk()) {
+                        logger.debug("handleCommand '{}' - returned Tivo Data Object: '{}'", myTivo.toString());
+                        updateTivoStatus(myTivo);
+                    }
+
+                    break;
+
+                case CHANNEL_TIVO_IRCMD:
+                    tivoCommand = "IRCODE";
+                case CHANNEL_TIVO_KBDCMD:
+                    logger.debug("handleCommand '{}' - CHANNEL_TIVO IRCODE/KBDCMD found!", getThing().getUID());
+
+                    if (tivoCommand == null) {
+                        tivoCommand = "KEYBOARD";
+                    }
+
+                    String tmpCommand = tivoCommand + " " + tmpAct;
+
+                    logger.debug("handleCommand '{}' - IR/KBD command to tivo: '{}'", getThing().getUID(), tmpCommand);
+
+                    // Attempt to execute the command on the TiVo
+                    myTivo = myTivoService.cmdTivoSend(tmpCommand);
+
+                    // Handle CHANNELUP and CHANNELDOWN which does not report a status change
+                    if (tmpAct == "CHANNELUP" | tmpAct == "CHANNELDOWN") {
+                        myTivoService.connTivoConnectRetry(false);
+                        myTivoService.connTivoConnectRetry(true);
```

Comment:
At one stage I thought that the CHANNELUP and CHANNELDOWN commands did not  return a status response.  So the disconnect and reconnect (the current channel is returned on new connection) was the only way to get this.  Turns out this does return a status, so this code is removed.